### PR TITLE
Implement graphQL api instead of REST for getting user repositories in github

### PR DESCRIPTION
**Pull request by tapishr**
- [x] Build succeeded
- [x] Manually Tested

## Summary by CodeRabbit

- **Refactor**
	- Enhanced the method for fetching user repositories to utilize GitHub's GraphQL API, improving data retrieval and pagination handling.

### CODE REVIEW

File: utils/providerAPI/getUserRepositories.ts [hunk 1]
```diff
@@ -64,37 +59,81 @@ type BitbucketRepoObj = {
...
+			const response = await axios.post(
+				'https://api.github.com/graphql',
+				{ query },
+				{
+					headers: {
+						Authorization: `Bearer ${access_token}`,
+					},
+				}
+			);
+			console.log(`[getUserRepositoriesForGitHub] response = ${JSON.stringify(response.data)}`)
```
Comment (avikalpg) [inline utils/providerAPI/getUserRepositories.ts]: remove console.log

File: utils/providerAPI/getUserRepositories.ts [hunk 2]
```diff
@@ -64,37 +59,81 @@ type BitbucketRepoObj = {
...
+
+			const data = response.data.data;
+			const repos: GithubRepoObj[] = data.viewer.repositories.nodes;
+
+			if (repos.length === 0) {
+				console.warn(`[getUserRepositories] No repositories received from GitHub API (provider-assigned id: ${authId}).`);
+				break;
+			}
```
Comment (avikalpg) [inline utils/providerAPI/getUserRepositories.ts:110]: Is this needed anymore?

File: utils/providerAPI/getUserRepositories.ts [hunk 3]
```diff
@@ -64,37 +59,81 @@ type BitbucketRepoObj = {
...
+
+			hasNextPage = data.viewer.repositories.pageInfo.hasNextPage;
+			endCursor = data.viewer.repositories.pageInfo.endCursor;
+		} catch (err) {
+			console.error(`[getUserRepositories] Error occurred while getting user repositories from GitHub GraphQL API (provider-assigned id: ${authId}).`, err);
+			throw err;
+		}
+	} while (hasNextPage);
+
+	// Sort repositories by repo_owner and then by repo_name
```
Comment (avikalpg) [inline utils/providerAPI/getUserRepositories.ts:128]: Low LOGAF

we might want to sort by activity or last updated as well. But this also looks good.

File: utils/providerAPI/getUserRepositories.ts [hunk 4]
```diff
@@ -7,15 +7,10 @@ import AuthInfo from "../../types/AuthInfo";
...
-	updated_at: string,
-	license: any,
-	private: boolean,
-	owner: object,
-	html_url: string,
-	[property: string]: any
+	name: string;
+	owner: {
+		login: string;
+	};
```
Comment (avikalpg) [inline utils/providerAPI/getUserRepositories.ts:13]: I think we should keep the `private` boolean here. We will use that very soon to display which repositories are free for use and which will cost them money.

**Review by avikalpg — APPROVED** [review]
I only have minor concerns or questions. This pull request looks good to me.

File: utils/providerAPI/getUserRepositories.ts (hunk 4 above)
Comment (MuskanPaliwal) [reply utils/providerAPI/getUserRepositories.ts:13]: I agree with @avikalpg 's point

File: utils/providerAPI/getUserRepositories.ts [hunk 5]
```diff
@@ -64,37 +59,79 @@ type BitbucketRepoObj = {
...
-		const allGitHubRepoIdentifiers = repos.map(repo => ({
-			repo_provider: supportedProviders[0],
-			repo_owner: repo.full_name.split('/')[0],
-			repo_name: repo.name
-		}))
-		allGitHubRepositories.push(...allGitHubRepoIdentifiers);
-		isResponseFull = repos.length == perPage;
-		pageNo++;
-	} while (isResponseFull);
+		return a.repo_owner.localeCompare(b.repo_owner);
```
Comment (MuskanPaliwal) [inline utils/providerAPI/getUserRepositories.ts:133]: just curious what does this localCompare() function do?

**Review by MuskanPaliwal — DISMISSED** [review]

**Pull request merged**